src/libraries/System.Private.CoreLib/src/System/Reflection/AssemblyNameLexer.cs

```diff
@@ -34,39 +30,89 @@ internal Token GetNext()
             return GetNext(out _);
         }
 
+        private static bool IsWhiteSpace(char ch)
+        {
+            switch (ch)
+            {
+                case '\n':
+                case '\r':
+                case ' ':
+                case '\t':
+                    return true;
+                default:
+                    return false;
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private char GetNextChar()
+        {
+            char ch;
+            if (_index < _input.Length)
+            {
+                ch = _input[_index++];
+                if (ch == '\0')
+                {
+                    throw new FileLoadException();
+                }
+            }
+            else
+            {
+                ch = '\0';
+            }
+
+            return ch;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private char PeekNextChar()
```

Comment:
I do not think we need this. The one use of it can be trivially replaced by `GetNextChar`